From: Jack Morgenstein <jackm@dev.mellanox.co.il>
Date: Sun, 19 Feb 2012 21:38:52 +0000
Subject: mlx4_core: Do not map BF area if capability is 0
Patch-mainline: v3.3-rc6
Git-commit: 3d7474734b220ccbf9997ea484d0bcd4f7ab8549
References: bnc#786036 FATE#314304

BF can be disabled in some cases, the capability field, bf_reg_size is set
to zero in this case. Don't map the BF area in this case, it would cause
failures.  In addition, leaving the BF area unmapped
also alerts the ETH driver to not use BF.

Signed-off-by: Jack Morgenstein <jackm@dev.mellanox.co.il>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Amir Vadai <amirv@mellanox.com>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/mellanox/mlx4/main.c |    3 +++
 1 file changed, 3 insertions(+)
--- a/drivers/net/ethernet/mellanox/mlx4/main.c
+++ b/drivers/net/ethernet/mellanox/mlx4/main.c
@@ -1036,6 +1036,9 @@ static int map_bf_area(struct mlx4_dev *
 	resource_size_t bf_len;
 	int err = 0;
 
+	if (!dev->caps.bf_reg_size)
+		return -ENXIO;
+
 	bf_start = pci_resource_start(dev->pdev, 2) +
 			(dev->caps.num_uars << PAGE_SHIFT);
 	bf_len = pci_resource_len(dev->pdev, 2) -
